src/Universum.hs

```diff
@@ -169,36 +174,76 @@ import           Lens.Micro               as X (Lens, Lens', Traversal, Traversa
                                                 (^?), _1, _2, _3, _4, _5)
 import           Lens.Micro.Mtl           as X (preuse, preview, use, view)
 
--- Type synonyms for lazy types
+-- For internal usage only
+import qualified Control.Exception.Base   (evaluate)
+import qualified Prelude                  (print)
+import qualified Text.Read                (readEither)
+
+-- | Type synonym for 'Data.Text.Lazy.Text'.
 type LText = Data.Text.Lazy.Text
+
+-- | Type synonym for 'Data.ByteString.Lazy.ByteString'.
 type LByteString = Data.ByteString.Lazy.ByteString
 
+-- | Renamed version of 'Prelude.id'.
 identity :: a -> a
 identity x = x
 
+-- | 'Prelude.map' generalized to 'Functor'.
 map :: Functor f => (a -> b) -> f a -> f b
 map = fmap
 
+-- | Alias for @flip map@.
```

Comment:
Perhaps let's remove `foreach`? It's not used anywhere in our code and it's unlikely that it ever will.